### PR TITLE
Time check

### PR DESCRIPTION
**Pull request by thawpt**
Addresses issue #768 
Did:
- Check to make sure start time is not after end time
- Made sure start time is not the same as the end time
- Added a clearer error message
Test:
- Create an event- make it recurring, make end time a time before the start time
- Once you try to create event, the error message "Event end time must be after start time." should pop up 
- Create an event- make it recurring, make end time the same as start time
- Once you try to create event, the error message "Event end time must be after start time." should pop up

### CODE REVIEW

**Review by AndersonStettner — CHANGES_REQUESTED** [review]
Make sure that if the event is not created due to one of the blockers (ie: the start and end time are the same) when the page is reloaded to show the error message the event is still marked as a recurring event.

**Comment by AndersonStettner** [comment]
Almost, after reload the recurring switch is still checked but the "End Date" box does not appear.
image

File: app/templates/admin/createEvent.html [hunk 1]
```diff
@@ -114,7 +115,8 @@ <h1 id="title">{{page_title}}</h1>
                     </div>
                 </div>
             </div>
-            <div class="form-group col-md-6 d-none" id ="endDateStyle">
+            {%if eventData.isRecurring == True and isNewEvent%}
```
Comment (AndersonStettner) [inline app/templates/admin/createEvent.html]: Set a variable to that will add or remove d-none depending on the condition so we don't have to repeat everything kind of like how we show and focus accordions.

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

File: app/templates/admin/createEvent.html (hunk 1 above)
Comment (sreynit02) [reply app/templates/admin/createEvent.html]: fixed!

**Comment by AndersonStettner** [comment]
[End Time is not checked when creating recurring events #768](https://trello.com/c/0s7oG47P/586-end-time-is-not-checked-when-creating-recurring-events-768)

File: app/static/js/createEvents.js [hunk 2]
```diff
@@ -34,16 +34,48 @@ function format24to12HourTime(timeStr){
...
+    calculateRecurringEventFrequency();
+  }
+
     $("#attachmentObject").fileinput({
         allowedFileExtensions:["pdf","jpg","png","gif", "csv", "docx", "jpg", "jpeg", "jfif"]
     })
   // Disable button when we are ready to submit
   $("#saveEvent").on('submit',function(event) {
-      $(this).find("input[type=submit]").prop("disabled", true)
+      debugger;
```
Comment (AndersonStettner) [inline app/static/js/createEvents.js]: Remove debug code

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

File: app/static/js/createEvents.js (hunk 2 above)
Comment (qasema) [reply app/static/js/createEvents.js]: @Andersonstettner removed

File: app/static/js/createEvents.js [hunk 3]
```diff
@@ -34,16 +34,47 @@ function format24to12HourTime(timeStr){
...
+      });
+
+  }
 /*
  * Run when the webpage is ready for javascript
  */
 $(document).ready(function() {
+  if ( $("#startDatePicker").val() && $("#endDatePicker").val()){
+    calculateRecurringEventFrequency();
+  }
```
Comment (AndersonStettner) [inline app/static/js/createEvents.js:70]: Remove this check, it does not have any actual use and it breaks the code.

**Review by AndersonStettner — CHANGES_REQUESTED** [review]

File: app/static/js/createEvents.js (hunk 3 above)
Comment (AndersonStettner) [reply app/static/js/createEvents.js:70]: Keep check but change it to check if the start and end dates are different instead

File: app/templates/admin/createEvent.html [hunk 4]
```diff
@@ -114,7 +115,14 @@ <h1 id="title">{{page_title}}</h1>
                     </div>
                 </div>
             </div>
-            <div class="form-group col-md-6 d-none" id ="endDateStyle">
+
+            {%if eventData.isRecurring == True and isNewEvent%}
```
Comment (Karina-Agliullova) [inline app/templates/admin/createEvent.html]: Add space after and before the "%" sign when using jinja in all jinja code

**Review by Karina-Agliullova — COMMENTED** [review]
Everything is fine. It just looks messy when spaces are not used in jinja code. so please include them. Other than that, good job!

**Comment by Karina-Agliullova** [comment]
Screenshot (1286)_LI
Screenshot (1287)_LI
When creating an event and setting it as recurring, after clicking save and saving it successfully, the end date option disappears. Also, if you check the saved recurring event on the events list page and try to edit one of the upcoming events, the end date option is also missing on the edit event page. Investigate why it occurs and fix the issue

**Review by Karina-Agliullova — APPROVED** [review]
Good work!

**Review by AndersonStettner — APPROVED** [review]
Looks good, good job!

**Pull request merged**